### DISCUSSION TITLE
SmartPorts

### DISCUSSION THREAD

**Discussion by christianbur** [CiscoDevNet/terraform-provider-meraki#20] [Ideas]
I would like to use the [Smartports ](https://documentation.meraki.com/MS/Port_and_VLAN_Configuration/SmartPorts)function, the function is currently in Early Access, the necessary [firmware MS-17.1.2](https://community.meraki.com/t5/Switching/MS-17-1-2-Released-and-Pushed-to-RC/m-p/246540) is already available as a release candidate. 

I need the function [organization-switch-ports-profiles,](https://developer.cisco.com/meraki/api-v1/create-organization-switch-ports-profile/) [organization-switch-ports-profiles-automation](https://developer.cisco.com/meraki/api-v1/create-organization-switch-ports-profiles-automation/) and a way to set them on the switch ports. 

Is there already a plan to implement this in this project?

**Comment by danischm** [comment]
The current implementation relies on the OpenAPI spec published here (https://github.com/meraki/openapi) which does not include early access API endpoints. I would need to check if we have one for those as well.

**Comment by danischm** [comment]
Added initial support for `meraki_switch_organization_ports_profile` here: https://github.com/CiscoDevNet/terraform-provider-meraki/commit/4aa0772adb3e3f6d713409cb363c31361e9c2c6f#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edR7

**Comment by danischm** [comment]
> ... and a way to set them on the switch ports.

Do you want to assign profiles just using automations or manually as well?

**Comment by danischm** [comment]
Added automations here: https://github.com/CiscoDevNet/terraform-provider-meraki/commit/b0bac132cb3654084677b400dc04c0060f525c0e

**Comment by danischm** [comment]
Resources have been added in v0.1.3 release.
